src/UserSettingsPage.cpp

```diff
@@ -286,112 +239,87 @@ UserSettingsPage::UserSettingsPage(QSharedPointer<UserSettings> settings, QWidge
-        encryptionLabel_->setFont(font);
+                formLayout_->addRow(label, layout);
+        };
 
-        auto general_ = new QLabel(tr("GENERAL"), this);
-        general_->setSizePolicy(QSizePolicy::Ignored, QSizePolicy::Fixed);
-        general_->setFont(font);
+        formLayout_->addRow(general_);
+        formLayout_->addRow(new HorizontalLine(this));
+        boxWrap(tr("Minimize to tray"), trayToggle_);
+        boxWrap(tr("Start in tray"), startInTrayToggle_);
+        formLayout_->addRow(new HorizontalLine{this});
+        boxWrap(tr("Circular Avatars"), avatarCircles_);
+        boxWrap(tr("Group's sidebar"), groupViewToggle_);
+        boxWrap(tr("Typing notifications"), typingNotifications_);
+        formLayout_->addRow(new HorizontalLine{this});
+        boxWrap(tr("Read receipts"), readReceipts_);
+        boxWrap(tr("Send messages as Markdown"), markdownEnabled_);
+        boxWrap(tr("Desktop notifications"), desktopNotifications_);
+        formLayout_->addRow(new QLabel("", this));
+        formLayout_->addRow(new HorizontalLine{this});
+        boxWrap(tr("Scale factor"), scaleFactorCombo_);
+        boxWrap(tr("Font size"), fontSizeCombo_);
+        formLayout_->addRow(new HorizontalLine(this));
+        boxWrap(tr("Font Family"), fontSelectionCombo_);
+        boxWrap(tr("Emoji Font Family"), emojiFontSelectionCombo_);
+        boxWrap(tr("Theme"), themeCombo_);
+        formLayout_->addRow(new QLabel("", this));
+        formLayout_->addRow(encryptionLabel_);
+        formLayout_->addRow(new HorizontalLine(this));
+        boxWrap(tr("Device ID"), deviceIdValue_);
+        boxWrap(tr("Device Fingerprint"), deviceFingerprintValue_);
+        formLayout_->addRow(new HorizontalLine(this));
+        formLayout_->addRow(sessionKeysLabel, sessionKeysLayout);
 
         mainLayout_ = new QVBoxLayout;
         mainLayout_->setAlignment(Qt::AlignTop);
         mainLayout_->setSpacing(7);
         mainLayout_->setContentsMargins(
           sideMargin_, LayoutTopMargin, sideMargin_, LayoutBottomMargin);
-        mainLayout_->addWidget(general_, 1, Qt::AlignLeft | Qt::AlignBottom);
-        mainLayout_->addWidget(new HorizontalLine(this));
-        mainLayout_->addLayout(trayOptionLayout_);
-        mainLayout_->addLayout(startInTrayOptionLayout_);
-        mainLayout_->addWidget(new HorizontalLine(this));
-        mainLayout_->addLayout(groupViewLayout);
-        mainLayout_->addLayout(avatarViewLayout);
-        mainLayout_->addWidget(new HorizontalLine(this));
-        mainLayout_->addLayout(typingLayout);
-        mainLayout_->addLayout(receiptsLayout);
-        mainLayout_->addLayout(markdownLayout);
-        mainLayout_->addLayout(desktopLayout);
-        mainLayout_->addWidget(new HorizontalLine(this));
+        mainLayout_->addLayout(formLayout_);
 
 #if defined(Q_OS_MAC)
-        scaleFactorLabel->hide();
+        // TODO: hide these with formlayout
```

Comment:
Maybe we could just not add them to the Layout? Specifically it sounds like we don't really need the scaleFactor and emojiFont rows on mac os.

Reply:
works for me, I'll take care of that asap